c++/include/orc/Reader.hh

```diff
@@ -288,6 +288,17 @@ namespace orc {
     virtual uint64_t getCompressionSize() const = 0;
 
     /**
+     * Get ID of writer that generated the file.
+     * Current availiable Orc writers:
+     * 0 = ORC Java
+     * 1 = ORC C++
+     * 2 = Presto
+     * @param id out parameter for writer id
+     * @return true if writer id is availiable, false if otherwise
+     */
+    virtual bool getWriterId(uint32_t & id) const = 0;
```

Comment:
Will there be any files generated by old Presto writer without an ID? Also if we choose to return ORC_FUTURE, there's no way to determine the actual writer ID.